projects/storefrontapp-e2e-cypress/cypress/helpers/anonymous-consents.ts

```diff
@@ -0,0 +1,254 @@
+import { registerUser, waitForPage } from './checkout-flow';
+import { checkBanner } from './homepage';
+import { signOutUser } from './login';
+import {
+  LANGUAGES,
+  LANGUAGE_DE,
+  LANGUAGE_LABEL,
+  LANGUAGE_REQUEST,
+  siteContextChange,
+  stub,
+} from './site-context-selector';
+import { generateMail, randomString } from './user';
+
+const ANONYMOUS_BANNER = 'cx-anonymous-consent-management-banner';
+const ANONYMOUS_DIALOG = 'cx-anonymous-consents-dialog';
+
+export function seeBannerAsAnonymous() {
+  cy.get(ANONYMOUS_BANNER).should('exist');
+}
+
+export function checkBannerHidden() {
+  cy.get(ANONYMOUS_BANNER).should('not.be.visible');
+}
+
+export function clickAllowAllFromBanner() {
+  cy.get(ANONYMOUS_BANNER)
+    .find('.btn-primary')
+    .click({ force: true });
+}
+
+export function clickViewDetailsFromBanner() {
+  cy.get(ANONYMOUS_BANNER)
+    .find('.btn-action')
+    .click({ force: true });
+}
+
+export function openDialogUsingFooterLink() {
+  cy.get('.anonymous-consents').within(() => {
+    const link = cy.get('a');
+    link.should('exist');
+    link.click({ force: true });
+  });
+}
+
+export function checkDialogOpen() {
+  cy.get(ANONYMOUS_DIALOG).should('exist');
+}
+
+export function checkDialogClosed() {
+  cy.get(ANONYMOUS_DIALOG).should('not.exist');
+}
+
+export function closeDialog() {
+  cy.get(`${ANONYMOUS_DIALOG} button.close`).click({ force: true });
+}
+
+export function giveOrRemoveAnonymousConsent(position) {
+  cy.get(`.cx-dialog-row:nth-child(${position})`)
+    .find('input')
+    .click({ force: true });
+}
+
+export function checkConsentsGiven() {
+  cy.get('.cx-toggle-text').each($match => {
+    cy.wrap($match).should('contain.text', 'ON');
+  });
+}
+
+export function checkConsentsWithdrawn() {
+  cy.get('.cx-toggle-text').each($match => {
+    cy.wrap($match).should('contain.text', 'OFF');
+  });
+}
+
+export function rejectAllConsent() {
+  cy.get(`${ANONYMOUS_DIALOG} .cx-action-link`)
+    .contains('Reject All')
+    .click({ force: true });
+}
+
+export function acceptAllConsent() {
+  cy.get(`${ANONYMOUS_DIALOG} .cx-action-link`)
+    .contains('Allow All')
+    .click({ force: true });
+}
+
+// export function loggedInUserBannerTest() {
+//   it('should not render the banner', () => {
+//     cy.get(ANONYMOUS_BANNER).should('not.exist');
+//   });
+// }
+
+// export function loggedInUserFooterLinkTest() {
+//   it('should not render the footer link', () => {
+//     cy.get('.anonymous-consents').should('not.exist');
+//   });
+// }
+
+export function sessionLogin() {
+  standardUser.registrationData.email = generateMail(randomString(), true);
+  cy.requireLoggedIn(standardUser);
+}
+
+export function moveAnonymousUserToLoggedInUser() {
+  it('should ignore the anonymous consents and load the previously given registered consents', () => {
+    cy.selectUserMenuOption({
+      option: 'Consent Management',
+    });
+
+    giveConsent();
+    signOutUser();
```

Comment:
Don't sign out from the consent management page. Navigate first to home page, and then sign out